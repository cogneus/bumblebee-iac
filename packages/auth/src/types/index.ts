export { TokenResponse } from './token-response.interface'
export { TokenServiceAttrs } from './token-service-attrs.interface'
