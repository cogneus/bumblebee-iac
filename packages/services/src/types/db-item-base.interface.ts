export interface DBItemPrimaryBase {
  PK: string
  SK: string
}
export interface DBItemSecondaryBase {
  PKGSI: string
  SKGSI: string
}
export interface DBItemBase extends DBItemPrimaryBase, DBItemSecondaryBase {}
