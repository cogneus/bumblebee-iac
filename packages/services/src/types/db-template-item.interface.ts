import { DBItemBase } from './db-item-base.interface'
import { TemplateVersion } from './template-version.interface'

export interface DBTemplateItem extends DBItemBase, TemplateVersion {}
