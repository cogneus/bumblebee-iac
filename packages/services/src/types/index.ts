export { DBTemplateItem } from './db-template-item.interface'
export { TemplateItem } from './template-item.interface'
export { TemplateVersion } from './template-version.interface'
export { QueryParams } from './query-params.interface'
export { ListParams } from './list-params.interface'
export { DBItemBase, DBItemPrimaryBase, DBItemSecondaryBase } from './db-item-base.interface'
