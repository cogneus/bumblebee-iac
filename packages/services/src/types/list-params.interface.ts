export interface ListParams {
  sort?: 'asc' | 'desc'
  includeCurrent?: boolean
  cursor?: string
  limit?: number
  fields?: string[]
}
