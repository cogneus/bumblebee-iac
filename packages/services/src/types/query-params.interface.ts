export interface QueryParams {
  sort?: 'asc' | 'desc'
  source?: string
  target?: string
  cursor?: string
  limit?: number
  fields?: string[]
}
