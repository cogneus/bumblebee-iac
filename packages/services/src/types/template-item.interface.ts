export interface TemplateItem {
  group: string
  source: string
  target: string
  template: Record<string, any>
}
