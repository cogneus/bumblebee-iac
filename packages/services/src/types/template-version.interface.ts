import { TemplateItem } from './template-item.interface'

export interface TemplateVersion extends TemplateItem {
  id: string
  version: number
  created: string
  modified: string
}
