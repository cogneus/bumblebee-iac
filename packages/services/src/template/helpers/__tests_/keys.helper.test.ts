import { TemplateVersion } from '../../../types'
import {
  makePK,
  makePKGSIGroup,
  makeSK,
  makeSKCurrent,
  makeSKGSIModified,
  removeKeyInfo,
} from '../keys.helper'

describe('Keys helper tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const item: TemplateVersion = {
    id: 'test-id',
    version: 1,
    created: '2023-01-01T00:00:00Z',
    modified: '2023-01-01T00:00:00Z',
    group: 'test-group',
    source: 'test-source',
    target: 'test-target',
    template: {},
  }

  test('makePK should match snapshot', () => {
    expect(makePK(item)).toMatchSnapshot()
  })

  test('makeSK should match snapshot', () => {
    expect(makeSK(item)).toMatchSnapshot()
  })

  test('makePKGSIGroup should match snapshot', () => {
    expect(makePKGSIGroup(item)).toMatchSnapshot()
  })

  test('makeSKCurrent should match snapshot', () => {
    expect(makeSKCurrent()).toMatchSnapshot()
  })

  test('makeSKGSIModified should match snapshot', () => {
    expect(makeSKGSIModified(item)).toMatchSnapshot()
  })

  test('removeKeyInfo should match snapshot', () => {
    expect(
      removeKeyInfo({
        PK: makePK(item),
        SK: makeSK(item),
        PKGSI: makePKGSIGroup(item),
        SKGSI: makeSKGSIModified(item),
        ...item,
      })
    ).toMatchSnapshot()
  })
})
