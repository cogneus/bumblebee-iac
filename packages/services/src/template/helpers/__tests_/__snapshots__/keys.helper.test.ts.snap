// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keys helper tests makePK should match snapshot 1`] = `"template#test-id"`;

exports[`Keys helper tests makePKGSIGroup should match snapshot 1`] = `"group#test-group"`;

exports[`Keys helper tests makeSK should match snapshot 1`] = `"version#0000000001"`;

exports[`Keys helper tests makeSKCurrent should match snapshot 1`] = `"current"`;

exports[`Keys helper tests makeSKGSIModified should match snapshot 1`] = `"modified#2023-01-01T00:00:00Z"`;

exports[`Keys helper tests removeKeyInfo should match snapshot 1`] = `
{
  "created": "2023-01-01T00:00:00Z",
  "group": "test-group",
  "id": "test-id",
  "modified": "2023-01-01T00:00:00Z",
  "source": "test-source",
  "target": "test-target",
  "template": {},
  "version": 1,
}
`;
